plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/Session.kt

```diff
@@ -90,22 +91,35 @@ class Session(val tabID: String, val project: Project) {
         AmazonqTelemetry.isApproachAccepted(amazonqConversationId = conversationId, enabled = true)
     }
 
+    suspend fun updateFilesPaths(
+        messenger: MessagePublisher,
+        tabId: String,
+        filePaths: List<NewFileZipInfo>,
+        deletedFiles: List<DeletedFileInfo>
+    ) {
+        messenger.updateFileComponent(tabId, filePaths, deletedFiles)
+    }
+
     /**
      * Triggered by the Insert code follow-up button to apply code changes.
      */
-    fun insertChanges(filePaths: List<NewFileZipInfo>, deletedFiles: List<String>, references: List<CodeReference>) {
+    fun insertChanges(filePaths: List<NewFileZipInfo>, deletedFiles: List<DeletedFileInfo>, references: List<CodeReference>) {
         val projectRootPath = context.projectRoot.toNioPath()
 
-        filePaths.forEach {
-            val filePath = projectRootPath.resolve(it.zipFilePath)
-            filePath.parent.createDirectories() // Create directories if needed
-            filePath.writeBytes(it.fileContent.toByteArray(Charsets.UTF_8))
-        }
+        filePaths
```

Comment:
if possible can we follow up on adding test for insertChanges. 
Same way we have here [link](https://github.com/aws/aws-toolkit-vscode/pull/4594)